[BUG] org.opensearch.index.IndexServiceTests.testAsyncTranslogTrimTaskOnClosedIndex flaky

Coming from https://github.com/opensearch-project/OpenSearch/pull/5412#issuecomment-1332638191


```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.index.IndexServiceTests.testAsyncTranslogTrimTaskOnClosedIndex" -Dtests.seed=B03404687BA373C2 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=es-US -Dtests.timezone=GB -Druntime.java=19

org.opensearch.index.IndexServiceTests > testAsyncTranslogTrimTaskOnClosedIndex FAILED
    java.lang.AssertionError: 
    Expected: <55L>
         but: was <110L>
        at __randomizedtesting.SeedInfo.seed([B03404687BA373C2:774AF1F12A370F5F]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:964)
        at org.junit.Assert.assertThat(Assert.java:930)
        at org.opensearch.index.IndexServiceTests.lambda$testAsyncTranslogTrimTaskOnClosedIndex$12(IndexServiceTests.java:450)
        at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1049)
        at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1022)
        at org.opensearch.index.IndexServiceTests.testAsyncTranslogTrimTaskOnClosedIndex(IndexServiceTests.java:449)
```

--- a/server/src/test/java/org/opensearch/index/IndexServiceTests.java
+++ b/server/src/test/java/org/opensearch/index/IndexServiceTests.java
@@ -421,7 +421,7 @@ public class IndexServiceTests extends OpenSearchSingleNodeTestCase {
         final String indexName = "test";
         IndexService indexService = createIndex(
             indexName,
-            Settings.builder().put(TRANSLOG_RETENTION_CHECK_INTERVAL_SETTING.getKey(), "100ms").build()
+            Settings.builder().put(TRANSLOG_RETENTION_CHECK_INTERVAL_SETTING.getKey(), "200ms").build()
         );
 
         Translog translog = IndexShardTestCase.getTranslog(indexService.getShard(0));

